src/Client.php

```diff
@@ -231,4 +231,11 @@ private function await(
             ->oneshot($channelId, $frameType)
             ->await($cancellation);
     }
+
+    public function __destruct()
+    {
+        if (\PHP_VERSION_ID >= 80400) {
+            $this->disconnect();
```

Comment:
Here we need to ensure that the connection has been established (`$this->connection !== null` or introduce a `state` enum), otherwise an exception will be thrown.

Reply:
Done.